src/components/ActionBar/ActionBar.js

```diff
@@ -14,11 +23,25 @@ const ActionBar = (props) => {
           id="readOnlyMode"
         ></Toggle>
       </div>
-      {props.isSomeCardChecked && (
-        <DeleteButton
-          deleteCardsHandler={props.deleteCardsHandler}
-        ></DeleteButton>
-      )}
+      <div className="action-bar__buttons">
+        {props.isSomeCardChecked && (
+          <DeleteButton
+            deleteCardsHandler={props.deleteCardsHandler}
+          ></DeleteButton>
+        )}
+        <Button kind="primary" onClick={updateModalState}>
+          <Add24></Add24> Add Card
+        </Button>
+      </div>
+      {isOpenState &&
+        ReactDOM.createPortal(
+          <AddCard
+            open={isOpenState}
+            onRequestClose={updateModalState}
+            onRequestSubmit={props.addNewCardHandler}
+          ></AddCard>,
+          document.getElementById("modal")
+        )}
```

Comment:
You're describing Modal window cases for Add and Delete actions in completely different ways, however that's the same case requring single concept.
Think about using common modal wrapper component to pass there required configurations. It can be used on a actionbar or even application level.